### DISCUSSION TITLE
error in dbt docs generate on Snowflake for multiple source objects when stages are not present in one's own snowflake schema

### DISCUSSION THREAD

**Discussion by shdwh** [ScalefreeCOM/datavault4dbt#225] [Q&A] [answered]
Dear all,

we are using datavault4dbt macros in dbt on Snowflake. We are using multiple source hubs, being fed from different stages.

We face a problem when running dbt docs when the person running this is not having ALL the stages in her/his Snowflake schema.

Do you have an idea / solution to fix this?

Output:
dbt docs generate
11:49:54  Running with dbt=1.6.12
11:49:54  Registered adapter: snowflake=1.6.0
11:50:36  Found 999 models, 1 operation, 3219 tests, 141 sources, 0 exposures, 0 metrics, 1350 macros, 0 groups, 0 semantic models
11:50:36  
11:50:37  Concurrency: 4 threads (target='dev')
11:50:37  
11:51:14  Encountered an error:
Runtime Error
  Compilation Error in model stg__sap__aufk (models/20_stage/sap/orders/stg__sap__aufk.sql)
    Could not find the derived_column input column COST_CENTER_BASIC_SETTLEMENT inside the source relation dev.myuserschema.raw__sap__aufk
    
    > in macro default__derived_columns_datatypes (macros/staging/derived_column_datatypes.sql)
    > called by macro derived_columns_datatypes (macros/staging/derived_column_datatypes.sql)
    > called by macro snowflake__stage (macros/staging/snowflake/stage.sql)
    > called by macro stage (macros/staging/stage.sql)
    > called by model stg__sap__de__030__aufk (models/20_stage/sap/orders/stg__sap__aufk.sql)

**Comment by tkirschke** [comment, marked as answer]
Hi @shdwh and thanks for reaching out!

This looks like a general dbt issue, might be similar to this issue: https://github.com/dbt-labs/dbt-core/issues/3741

To solve, first do a compile against a target where all objects are available, then do a 
`dbt docs generate --no-compile`

Let me know if this helps!

Best regards
Tim

**Comment by shdwh** [comment]
Hi @tkirschke,
this helps thank you very much.
Best regards
Sandra
